accounts/abi/bind/template.go

```diff
@@ -106,7 +106,7 @@ var (
 		  if err != nil {
 		    return common.Address{}, nil, nil, err
 		  }
-		  return address, tx, &{{.Type}}{ {{.Type}}Caller: {{.Type}}Caller{contract: contract}, {{.Type}}Transactor: {{.Type}}Transactor{contract: contract}, {{.Type}}Filterer: {{.Type}}Filterer{contract: contract} }, nil
```

Comment:
Is there a reason for renaming `contract` to `BoundContract`? I'm not a big fan of this, since people might regenerate their bindings in order to bring in newer features, and then they potentially have to rewrite their entire codebase.